### ISSUE TITLE
Minor bugs in ImageNet training tutorial

### ISSUE THREAD

**Issue by youn123** [apache/incubator-mxnet#15742]
Not sure if this is the correct place to point this out, but the ImageNet training tutorial seems to contain a few minor bugs. Fixing them might save people some time:

In [ImageNet data preparation tutorial](https://gluon-cv.mxnet.io/build/examples_datasets/imagenet.html), I added these to `extract_train` of `imagenet.py` at lines 82-83:

```
if clas_tar.name == '.' or class_tar.name == '..':
    continue
```

Also, to extract the data that is directly usable in the training tutorial, user should run:

```
python imagenet.py --download-dir ~/ILSVRC2012 --with-rec
```

So call the script with `--with-rec` set since it's not set by default.

In [ImageNet training tutorial](https://gluon-cv.mxnet.io/build/examples_classification/dive_deep_imagenet.html):

- a `batch_size` variable should be defined globally
- `lr_decay_period` should be changed to `lr_decay_count`
- `loss = [L(yhat, y) for yhat, y in zip(outputs, label)]` `L` should be changed to `loss_fn`
- `opt.log_interval` should change to just `log_interval`

**Comment by vrakesh** [comment]
@youn123 Thank you for pointing this out

@mxnet-label-bot add [Gluon, Website]

**Comment by ys2843** [comment]
This issue is not relating to MXNet website, remove "Website" label.
@mxnet-label-bot remove [Website]
